### PR TITLE
Fix fetching plenty of branches in pipeline

### PR DESCRIPTION
**Pull request by gitlawr**
Address issue: https://github.com/rancher/rancher/issues/17231

Problem:
    1. Fails to list all branches when a pipeline repository has a large number of branches(>100).
    2. It takes a long time to fetch branches and pipeline files when there are plenty of branches.

Solution:
    1. Handle pagination.
    2. Update branches link to get all branches. So that we can reduce API calls from n+1 to 2, one for list branches and one for getting the pipeline file in a specific branch on demand.

### CODE REVIEW

**Comment by orangedeng** [comment]
LGTM

File: pkg/pipeline/remote/bitbucketcloud/client.go [hunk 1]
```diff
@@ -257,19 +257,21 @@ func (c *client) GetBranches(repoURL string, accessToken string) ([]string, erro
...
+		b, err := getFromBitbucket(nexturl, accessToken)
+		if err != nil {
+			return nil, err
+		}
+		var pageBranches PaginatedBranches
+		if err := json.Unmarshal(b, &pageBranches); err != nil {
+			return nil, err
+		}
+		for _, branch := range pageBranches.Values {
+			result = append(result, branch.Name)
```
Comment (cbron) [inline pkg/pipeline/remote/bitbucketcloud/client.go]: Do you not need the `if b.Type != "BRANCH" ` check happening in Bitbucket server ?

Comment (gitlawr) [reply pkg/pipeline/remote/bitbucketcloud/client.go]: It is a redundant check, given that the branches API always returns branch objects. Updated and removed the check.

**Comment by gitlawr** [comment]
Updated

**Pull request merged**